src/markdoc/nodes/Image.svelte

```diff
@@ -1,17 +1,96 @@
 <script lang="ts">
-	import { getContext, hasContext } from 'svelte';
+    import Tooltip from '$lib/components/Tooltip.svelte';
+    import { createDialog, createTooltip, melt } from '@melt-ui/svelte';
+    import { getContext, hasContext } from 'svelte';
+    import { fade, scale } from 'svelte/transition';
 
-	export let src: string;
-	export let alt: string;
-	export let title: string;
+    export let src: string;
+    export let alt: string;
+    export let title: string;
 
-	const inTable = hasContext('in-table') ? getContext('in-table') : false;
+    const inTable = hasContext('in-table') ? getContext('in-table') : false;
+
+    const {
+        elements: { portalled, trigger, content, overlay },
+        states: { open }
+    } = createDialog({
+        preventScroll: false,
+        forceVisible: true
+    });
 </script>
 
 {#if inTable}
-	<img {src} {alt} {title} loading="lazy" style:vertical-align="middle" />
+    <img {src} {alt} {title} loading="lazy" style:vertical-align="middle" />
 {:else}
-	<div class="aw-media">
-		<img {src} {alt} {title} loading="lazy" class="aw-u-media-ratio-16-9 u-width-full-line" />
-	</div>
+    <div class="aw-media">
+        <img {src} {alt} {title} loading="lazy" class="aw-u-media-ratio-16-9 u-width-full-line" />
+
+        <div class="abs">
+            <Tooltip closeOnPointerDown>
+                <button class="aw-button is-secondary" use:melt={$trigger}>
+                    <span class="icon-arrow-expand" aria-hidden="true" />
+                </button>
+                <svelte:fragment slot="tooltip">Expand</svelte:fragment>
+            </Tooltip>
+        </div>
+    </div>
+
+    {#if $open}
+        <div use:melt={$portalled}>
```

Comment:
this goes offscreen
<img width="1440" alt="Bildschirmfoto 2023-10-18 um 13 04 16" src="https://github.com/appwrite/website/assets/1759475/fd1217c1-b4fd-4216-9dda-77c35caa103c">